tests/test_fuzzy_checker.py

```diff
@@ -1,7 +1,248 @@
-# TODO: Add tests for fuzzy_checker
+from __future__ import annotations
+
+from typing import TYPE_CHECKING
+
+import numpy as np
+import pandas as pd
 import pytest
+import scipy
+from _pytest.logging import LogCaptureFixture
+
+if TYPE_CHECKING:
+    from py._path.local import LocalPath
+
+from vivarium_testing_utils.fuzzy_checker import FuzzyChecker
+
+OBSERVED_DENOMINATORS = [100_000, 1_000_000, 10_000_000]
+TARGET_PROPORTION = 0.1
+LOWER_BOUNDS = [
+    1e-14,
+    1e-10,
+    0.000001,
+    0.01,
+    0.1,
+    0.2,
+    0.3,
+    0.4,
+    0.5,
+    0.6,
+    0.8,
+    0.9,
+    0.99999,
+    1.0 - 1e-10,
+]
+WIDTHS = [
+    1e-14,
+    1e-12,
+    1e-10,
+    0.0000001,
+    0.00001,
+    0.001,
+    0.01,
+    0.03,
+    0.05,
+    0.1,
+    0.2,
+    0.4,
+    0.6,
+    0.9,
+]
+
+
+@pytest.mark.parametrize(
+    "numerator, denominator, target_proportion",
+    [(10_008, 100_000, 0.1), (976, 1_000_000, 0.001), (1_049, 50_000, (0.0198, 0.0202))],
+)
+def test_pass_fuzzy_assert_proportion(
+    numerator: int, denominator: int, target_proportion: float
+) -> None:
+    FuzzyChecker().fuzzy_assert_proportion(numerator, denominator, target_proportion)
+
+
+@pytest.mark.parametrize(
+    "numerator, denominator, target_proportion, match",
+    [
+        (901, 100_000, 0.05, "is significantly less than expected"),
+        (1_150, 50_000, 0.02, "is significantly greater than expected"),
+    ],
+)
+def test_fail_fuzzy_assert_proportion(
+    numerator: int, denominator: int, target_proportion: float, match: str
+) -> None:
+    with pytest.raises(AssertionError, match=match):
+        FuzzyChecker().fuzzy_assert_proportion(numerator, denominator, target_proportion)
+
+
+def test_small_sample_size_fuzzy_assert_proportion(caplog: LogCaptureFixture) -> None:
+    FuzzyChecker().fuzzy_assert_proportion(1, 10, 0.1)
+    assert "Sample size too small" in caplog.text
+
+
+def test_not_conclusive_fuzzy_assert_proportion(caplog: LogCaptureFixture) -> None:
```

Comment:
nit: function docstrings - as opposed to random inline comments - should be in """ (throughout this file)